pub/lib/dependabot/pub/file_updater.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "dependabot/file_updaters"
+require "dependabot/file_updaters/base"
+require "dependabot/pub/helpers"
+
+module Dependabot
+  module Pub
+    class FileUpdater < Dependabot::FileUpdaters::Base
+      include Dependabot::Pub::Helpers
+
+      def initialize(dependencies:, dependency_files:, repo_contents_path: nil,
+                     credentials:, options: {}, pub_hosted_url: nil)
+        @pub_hosted_url = pub_hosted_url
+        super(
+          dependencies: dependencies,
+          dependency_files: dependency_files,
+          repo_contents_path: repo_contents_path,
+          credentials: credentials,
+          options: options
+        )
+      end
+
+      def self.updated_files_regex
+        [
+          /^pubspec\.yaml$/,
+          /^pubspec\.lock$/
+        ]
+      end
+
+      def updated_dependency_files
+        dependency_services_apply(@dependencies)
```

Comment:
Is it possible for a dependency update to trigger executing code provided by that dependency? I think the answer is currently no but wanted to confirm. If it is we'll put some additional restrictions in place regarding private repo/registry access (which won't affect anything here since it's not supported yet).

Examples of this behavior in other ecosystems:
- https://medium.com/ochrona/arbitrary-code-execution-during-python-package-installation-3a60990350ef
- https://medium.com/@kyle_martin/understanding-and-protecting-against-malicious-npm-package-lifecycle-scripts-8b6129619d7c

Reply:
No, that would be a bug in pub.